src/header/NamespaceDropdown/NamespaceDropdown.tsx

```diff
@@ -1,18 +1,49 @@
 import { Menu, Icon } from 'fundamental-react';
-import { Link } from 'react-router-dom';
+import { useRecoilValue } from 'recoil';
+import { Link, useLocation } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
 
+import { namespacesState } from 'state/namespacesAtom';
 import { useUrl } from 'hooks/useUrl';
-import { NamespacesState } from 'state/namespacesAtom';
 
 import './NamespaceDropdown.scss';
 
+const Namespaces = () => {
+  const namespaces = useRecoilValue(namespacesState);
+  const { clusterUrl, namespace, namespaceUrl } = useUrl();
+  const { pathname } = useLocation();
+
+  const switchNamespace = (namespaceName: string) => {
+    if (!pathname.includes('/namespaces/'))
+      return clusterUrl(`namespaces/${namespaceName}`);
+
+    const pathElements = pathname
+      .substring(pathname.indexOf('namespaces/'))
+      .split('/');
+
+    const resourceType = pathElements[2];
+    const resourceName = pathElements[3];
+
+    if (!resourceName) return pathname.replace(namespace, namespaceName);
```

Comment:
is the `resourceName` handler even needed here?
it seems to work just fine with just the `namespaceUrl` below